src/dataflow/src/render/sinks.rs

```diff
@@ -226,6 +227,10 @@ where
                     }
                 }
 
+                let active_write_worker =
+                    (usize::cast_from(sink_id.hashed()) % scope.peers()) == scope.index();
```

Comment:
thought about this for a while and couldn't think of a better thing to do in the short term, added a comment

Reply:
Please feel free to ignore because I don't want to derail the PR, but: couldn't we just have the `shared_frontier` for all workers. And for those workers that are not writing, it will just mean that no updates are happening but they're also not blocking anything? Could be that the idle workers (for that sink) have to advance the frontier to "max" before they quit.